### PR TITLE
IDEM-106

### PR DESCRIPTION
**Pull request by JasminFarhi**
Image from iOS (117)
Image from iOS (118)

### CODE REVIEW

**Review by sbm-peter — APPROVED** [review]

**Review by jrocca82 — DISMISSED** [review]

File: app/src/utils/formatter.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const reformatDate = (date: string) => {
+  if (date) {
+    const dateArray = date.split("");
+    const day = Number(dateArray.slice(0, 2).join(""));
+    const month = dateArray.slice(3, 5).join("");
+    const year = Number(dateArray.slice(6, 10).join(""));
```
Comment (bitcoinbrisbane) [inline app/src/utils/formatter.ts:6]: There is no need to do a split.  Use a .substring

File: app/src/utils/birthday-utils.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { ClaimData } from "../types/claim";
+import moment from "moment";
+import { reformatDate } from "./formatter";
+
+export const check18Plus = (claim: ClaimData): boolean => {
+  if (claim.type === "DateOfBirthCredential") {
+    const userBirthday = reformatDate(claim.value.dob);
+    const age = moment(userBirthday, "YYYYMMDD").fromNow();
+    const userAge = age.split("").slice(0, 2).join("");
```
Comment (bitcoinbrisbane) [inline app/src/utils/birthday-utils.ts:9]: Use a .substring

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Don't convert to an array.

File: app/src/utils/birthday-utils.ts (hunk 2 above)
Comment (bitcoinbrisbane) [reply app/src/utils/birthday-utils.ts:9]: We also have a year of birth utils.  We could have used that to get the year of birth.

File: app/src/utils/birthday-utils.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    if (age.includes("in")) {
+      return false;
+    }
+    if (Number(userAge) >= 18) {
+      return true;
+    }
```
Comment (bitcoinbrisbane) [inline app/src/utils/birthday-utils.ts:15]: This method doesn't make sense.  We can use the functions of the moment.js lib to do all this for us.

File: app/src/utils/formatter.ts (hunk 1 above)
Comment (bitcoinbrisbane) [reply app/src/utils/formatter.ts:6]: This method could be done by using moment to parse the date.  Then a .toString with the specific format required.

**Review by bitcoinbrisbane — DISMISSED** [review]
Use moment js instead of rolling your own!

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 4]
```diff
@@ -222,9 +269,10 @@ const ClaimScreen: React.FC = () => {
                   <Text style={{ marginBottom: 20 }}>{field.title}</Text>
                   <Switch
                     value={formState[field.id] === "true"}
-                    onValueChange={(value) =>
-                      onChange(value ? "true" : "false")
-                    }
+                    onValueChange={(value) => {
+                      onChange(value ? "true" : "false");
+                      console.log();
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/ClaimScreen.tsx:274]: remove?

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 5]
```diff
@@ -70,6 +70,53 @@ const ClaimScreen: React.FC = () => {
...
+            }
+          );
+        }
+      }
+      return null;
+    });
+    setLoading(false);
+    return findBirthday;
+  };
+
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/ClaimScreen.tsx:119]: lets put logic into hook please

**Review by xiangzhe0812 — COMMENTED** [review]
Nice work, only two small stuff

File: app/src/hooks/useClaimScreen.ts [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+type Hooks = {
+  loading: boolean;
+  saveAndCheckBirthday: () => void;
+  onSelectFile: (fileId: string) => void;
+  selectedFileIds: string[];
+  setLoading: (loading: boolean) => void;
+  setSelectedFileIds: (selectedFileIds: string[]) => void;
+  verifyEmail: (email: string) => void;
+};
+const useClaimScreen = (): Hooks => {
```
Comment (bitcoinbrisbane) [inline app/src/hooks/useClaimScreen.ts:19]:
```suggestion
};

const useClaimScreen = (): Hooks => {
```

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**